gitpod-io/gitpod
# New User Onboarding Flow UI

## Description

This adds a new onboarding flow for new users (currently we consider new users anyone that hasn't provided a preferred IDE). For those users, we'll show the following three step flow when they sign in.

| Step One      | Step Two | Step Three
| ----------- | ----------- | ----------- |
| <img width="772" alt="image" src="https://user-images.githubusercontent.com/367275/220799436-7e6bdf64-e25a-415f-a124-32d88e5ccb47.png"> | <img width="777" alt="image" src="https://user-images.githubusercontent.com/367275/220799682-cc5c7a55-1180-46a4-ae76-b5a3d50ed483.png"> | <img width="682" alt="image" src="https://user-images.githubusercontent.com/367275/220799747-23c6e93a-6ae1-4bf1-8efd-f1a0af2c99ce.png">

* This is behind the `newSignupFlow` feature flag that is enabled in non-production.
* While we're collecting a distinct First and Last name in the flow, we're still storing these as a unified `fullName` on the user. Collecting as 2 fields is an attempt to help signal we're asking for real names and not aliases/usernames.
* Email is still stored on `user.additionalDetails.profile.emailAddress`
* Answers to step 3 are stored on new fields under `user.additionalDetails.profile`
* At the completion of the onboarding flow we're storing a timestamp on `user.additionalDetails.profile.onboardedTimestamp` to help us identify which user's have completed this flow in the future. We plan to introduce this flow, or a variation of it to existing users, and this will help us distinguish those that have gone through it.

## Related Issue(s)

Fixes #

## How to test

* Visit the preview environment: https://bmh-onboarding-form.preview.gitpod-dev.com/
* The first time you visit, you should be sent through the flow.
* You can verify your name and email changes are applied after completing the flow by visiting the User Settings.

**Repeat the flow**
You can also initiate the onboarding flow manually by appending the query param `?onboarding=force` to a url. For example: 

https://bmh-onboarding-form.preview.gitpod-dev.com/workspaces?onboarding=force 

... should kick you into the flow even if you've completed it. This was mainly added to aid in testing, but could be helpful in the future to send a user into the flow manually.

**With Starting Workspace links:**

You can also test this with a workspace context link and include the `?onboarding=force` query param:

https://bmh-onboarding-form.preview.gitpod-dev.com/?onboarding=force#https://github.com/gitpod-io/gitpod

## Release Notes

```release-note
We've introduced a new onboarding flow to help user's customize Gitpod and get to know their needs better.
```

## Documentation


## Build Options:

      Experimental feature to run the build with GitHub Actions (and not in Werft).
      leeway-target=components:all
      Run Leeway with `--dont-test`

<details>
<summary>Publish Options</summary>

</details>

<details>
<summary>Installer Options</summary>

  Add desired feature flags to the end of the line above, space separated
</details>

#### Preview Environment Options:
      If enabled this will build `install/preview`
      If enabled this will create the environment on GCE infra
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`
